scrapers_next/wa/people.py

```diff
@@ -0,0 +1,167 @@
+import re
+from spatula import HtmlListPage, CSS, XPath, URL
+from ..common.people import ScrapePerson
+
+
+class PartyAugmentation(HtmlListPage):
+    """
+    WA Email addresses are listed on a separate page.
+    """
+
+    source = URL("https://app.leg.wa.gov/memberemail/Default.aspx")
+
+    def find_rows(self):
+        for table in CSS("#membertable").match(self.root):
+            rows = CSS("tbody tr").match(table)
+            if len(rows) == 147:
+                return rows
+
+    def process_page(self):
+        # We need it to find the member's email address.
+        # These positions are enough to discriminate the chamber too (0 = upper, 1,2 = lower)
+        mapping = {}
+        rows = self.find_rows()
+        for row in rows:
+            tds = row.getchildren()
+            name = (
+                CSS("a")
+                .match_one(tds[0])
+                .text_content()
+                .lstrip(r"^(Rep\.|Senator)")
+                .strip()
+            )
+            email = tds[1].text_content().strip()
+            dist = tds[2].text_content().strip()
+            position = tds[3].text_content().strip()
+            party = tds[4].text_content().strip()
+            mapping[name] = (email, party, dist, position)
+        return mapping
+
+
+class LegList(HtmlListPage):
+    dependencies = {"party_mapping": PartyAugmentation()}
+
+    def process_item(self, item):
+        title_name_party = XPath('.//span[@class="memberName"]/text()').match_one(item)
+
+        (name, party) = re.search(
+            r"^(?:Senator|Representative)\s(.+)\s\(([RD])\)$", title_name_party
+        ).groups()
+        if party == "R":
+            party = "Republican"
+        elif party == "D":
+            party = "Democratic"
+
+        (district_name, _district_name) = XPath(
+            ".//a[contains(text()," ' " Legislative District")]/text()'
```

Comment:
this is a bit tough to read, and can be cleaned up by using 
`
            './/a[contains(text(), "Legislative District")]/text()'
`

instead of the two split strings